salt/_modules/metalk8s_kubernetes.py

```diff
@@ -2728,3 +2728,29 @@ def show_api_service(name, **kwargs):
             raise CommandExecutionError(exc)
     finally:
         _cleanup(**cfg)
+
+
+def read_storage_class(name, **kwargs):
```

Comment:
Why would we need this? All required info should be in the pillar.

Reply:
@NicolasT I am assuming here that at the start of execution we do not refresh pillar and for some reason, a volume in the pillar has a storage class which no longer exist(volume storageclass was deleted).

`But again, I think pillars get refreshed every now and then`

So,
- We either refresh the pillars and make sure that all volumes in the pillars actually have a valid storage class
- Or Check that each volume to be provisioned has a storage class indeed(reason for the above method).

Reply:
Volumes in the pillar always have a valid storageclass, right? That's checked in the `ext_pillar`.

The main reason I was asking is as follows: I assume you're adding this code to eventually be used in some state and a state SLS. These SLSs are to be applied on minions, which is in the end where the storage sits.

However.

This code connects to the K8s API. And minions can't connect to the K8s API. Hence all info a minion needs that's derived from info that's stored in the K8s API needs to be passed in as pillar values.

Reply:
Thanks for the clarification. It will be useful to simply refresh the pillars hence ensure that all volumes in the pillar are valid(have storageclasses).